spec/index.spec.js

```diff
@@ -369,3 +371,27 @@ describe('isIdempotentRequestError(error)', () => {
     expect(isIdempotentRequestError(errorResponse)).toBe(false);
   });
 });
+
```

Comment:
A test for the delay implementation itself is needed.

Reply:
Do you have a pointer of how to test this? 

Given the following:

```js
const client = axios.create();

setupResponses(client, [
  () => nock('http://example.com').get('/test').replyWithError(NETWORK_ERROR),
  () => nock('http://example.com').get('/test').replyWithError(NETWORK_ERROR),
  () => nock('http://example.com').get('/test').replyWithError(NETWORK_ERROR),
  () => nock('http://example.com').get('/test').reply(200, 'It worked!')
]);
  
axiosRetry(client, { 
  retries: 4,
  retryCondition: () => true,
  retryDelay : axiosRetry.exponentialDelay
});

client.get('http://example.com/test').then(result => {
  expect(result.status).toBe(200);
  done();
}, done.fail);
```

How could I verify the delays between requests are heppening?

Reply:
Maybe easiest solutions would be to store new Date.now() before first request, and then compare time after request is done. (if it's in specified random limits)

Reply:
Thanks, I was thinking along those lines, but was not sure if there was a way that did not rely on timing the execution.

Reply:
You can try mock timers, there is for sure mock for mocha.

Reply:
@moose56 anyway, is it needed to check exponential delay this way? Function is already tested (what values it generates). And if retryStrategy is working, you could pass custom retry strategy to the config and have just timeout 1ms, and eg add there counter `let retryCount = 0` and in strategy `retryCount += 1`.

Reply:
Ah, OK so something like:

```js
const client = axios.create();

setupResponses(client, [
  () => nock('http://example.com').get('/test').replyWithError(NETWORK_ERROR),
  () => nock('http://example.com').get('/test').replyWithError(NETWORK_ERROR),
  () => nock('http://example.com').get('/test').replyWithError(NETWORK_ERROR),
  () => nock('http://example.com').get('/test').reply(200, 'It worked!')
]);
  
let retryCount = 0;

axiosRetry(client, { 
  retries: 4,
  retryCondition: () => true,
  retryDelay : () => { retryCount += 1; return 0; }
});

client.get('http://example.com/test').then(result => {
  expect(retryCount).toBe(3);
  done();
}, done.fail);
```